### PR TITLE
Huge update

### PR DESCRIPTION
**Pull request by ghost**
What the hell this PR does?!

-Adding BLAKE3,BLAKE2S,BLAKE2B support, as well as switching some of crates to the RustCrypto maintaines ones the blake2 and md5 crate
-Update to Rust 2018
-Fix a lot of clippy warns

That's all for now :)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,32 +10,33 @@ license = "MIT"
...
 [dependencies]
-futures-cpupool = "0.1"
-blake2-rfc = "0.2"
-once_cell = "1.5"
-tabwriter = "1.2"
-num_cpus = "1.13"
-futures = "0.1"
-walkdir = "2.3"
-crc32c = "0.5"
+futures-cpupool = "0.1.8"
```
Comment (nabijaczleweli) [inline Cargo.toml:18]: sort these reverse-length and drop the last version component, like it was before

File: src/algorithms.rs [hunk 2]
```diff
@@ -71,8 +73,10 @@ impl Algorithm {
             Algorithm::SHA2512 |
             Algorithm::SHA3512 |
             Algorithm::BLAKE |
-            Algorithm::BLAKE2 |
             Algorithm::MD6512 => 128,
+            Algorithm::BLAKE2B => 128,
+            Algorithm::BLAKE2S => 64,
+            Algorithm::BLAKE3 => 64,
```
Comment (nabijaczleweli) [inline src/algorithms.rs:79]: merge these with the rest like it was before

File: src/algorithms.rs [hunk 3]
```diff
@@ -90,7 +94,9 @@ impl FromStr for Algorithm {
             "sha3256" | "sha3-256" | "sha-3-256" => Ok(Algorithm::SHA3256),
             "sha3" | "sha-3" | "sha3512" | "sha3-512" | "sha-3-512" => Ok(Algorithm::SHA3512),
             "blake" => Ok(Algorithm::BLAKE),
-            "blake2" => Ok(Algorithm::BLAKE2),
+            "blake2b" => Ok(Algorithm::BLAKE2B),
```
Comment (nabijaczleweli) [inline src/algorithms.rs:97]: blake2 => blake2b too

File: src/hashing/mod.rs [hunk 4]
```diff
@@ -71,7 +73,9 @@ pub fn hash_reader<R: Read>(data: &mut R, algo: Algorithm) -> String {
         Algorithm::SHA3256 => sha3256_3512::sha3256::hash(data),
         Algorithm::SHA3512 => sha3256_3512::sha3512::hash(data),
         Algorithm::BLAKE => blake::hash(data),
-        Algorithm::BLAKE2 => blake2::hash(data),
+        Algorithm::BLAKE2B => blake2b::hash(data),
```
Comment (nabijaczleweli) [inline src/hashing/mod.rs:76]: likewise

File: src/options.rs [hunk 5]
```diff
@@ -53,7 +53,7 @@ impl Options {
             .about("Tool for making/verifying checksums of directory trees")
             .args(&[Arg::from_usage("[DIRECTORY] 'Directory to hash/verify'").default_value(".").validator(Options::directory_validator),
                     Arg::from_usage("--algorithm=[algorithm] -a 'Hashing algorithm to use. {n}\
-                                     Supported algorithms: SHA{1,2-{224,256,384,512},3-{256,512}}, BLAKE{,2}, CRC{64,32{,C},16,8}, MD{5,6-{128,256,512}}, XOR8'")
+                                     Supported algorithms: SHA{1,2-{224,256,384,512},3-{256,512}}, BLAKE{,2B,2S,3}, CRC{64,32{,C},16,8}, MD{5,6-{128,256,512}}, XOR8'")
```
Comment (nabijaczleweli) [inline src/options.rs:55]: `BLAKE{,2=2B,2S,3)`

**Review by nabijaczleweli — CHANGES_REQUESTED** [review]
Drop everything from the second patch onward, otherwise looks mostly fine, see comments inline and update the manpage

**Comment by ghost** [comment]
Why not to update to the latest Rust edition?

**Comment by nabijaczleweli** [comment]
Because it's not available yet. As for 2018: because it's not very good.

File: Cargo.toml (hunk 1 above)
Comment (ghost) [reply Cargo.toml:18]: Eh... Can you explain a bit more? I don't get it

**Comment by ghost** [comment]
> Because it's not available yet. As for 2018: because it's not very good.

I was about 2018. But why 2018 version is bad?

File: Cargo.toml (hunk 1 above)
Comment (nabijaczleweli) [reply Cargo.toml:18]: These are sorted length-decreasing then alphabetical: keep that ordering. And remove the patch version component from the specs.

**Comment by nabijaczleweli** [comment]
Too many bad layout opinions and implicit crate imports, among others

**Comment by ghost** [comment]
Do you think Rust 2021 gonna be better?

**Comment by nabijaczleweli** [comment]
That's neither here nor there, and I don't really care until I use it

**Comment by nabijaczleweli** [comment]
Released in [`v0.8.0`](https://github.com/nabijaczleweli/checksums/releases/tag/v0.8.0)

**Pull request closed without merging**